hawkular-wildfly-agent-installer/src/main/java/org/hawkular/wildfly/agent/installer/AgentInstaller.java

```diff
@@ -349,8 +352,21 @@ private static String createSecurityRealm(String keystoreFile, String keystorePa
      */
     private static XmlEdit createStorageAdapter(boolean withHttps, InstallerConfiguration installerConfig) {
         String select = "/server/profile/*[namespace-uri()='urn:org.hawkular.agent:agent:1.0']/";
-        StringBuilder xml = new StringBuilder("<storage-adapter")
-                .append(" type=\"HAWKULAR\"");
+        StringBuilder xml = new StringBuilder("<storage-adapter");
+
+        String tenantId = installerConfig.getTenantId();
+        if (installerConfig.isMetricsOnlyMode()) {
+            xml.append(" type=\"METRICS\"");
+            if (tenantId == null || tenantId.isEmpty()) {
+                throw new IllegalArgumentException("You must specify tenant-id when in metrics-only mode");
+            }
+        } else {
+            xml.append(" type=\"HAWKULAR\"");
+        }
+
+        if (tenantId != null && !tenantId.isEmpty()) {
+            xml.append(" tenantId=\"" + tenantId + "\"");
```

Comment:
From what I'm understanding on this code, it allows to include a tenantId on a non-metrics type as well. I think this should *not* be allowed, as it breaks the user's expectations: this is either ignored later on, or is used and the backend rejects the request.